### PR TITLE
Tgc 432 xstate

### PR DESCRIPTION
**Pull request by Andrew-Folga**

### CODE REVIEW

File: docker-compose.test.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: '3.8'
+version: "3.8"
```
Comment (alanplatt) [inline docker-compose.test.yaml]: needs removing

File: src/config/machines/example-grant-machine.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+     * @returns {string} event's question
+     */
+    currentPageId: (context, event) => {
+      return event.nextPageId;
+    },
+  }),
+};
+
+const exampleGrantMachine = createMachine({
+  /** @xstate-layout N4IgpgJg5mDOIC5RgB4EMC2AHANmA4gE5oB2ALgLJoDGAFgJYlgB0sZahZAxAHICiADQAqAbQAMAXUSgsAe1j0y9WSWkgUiAIwBmAKzMAbJoCcAdgAcB0wd0AWIwdsAaEAE8t2sc1vaATNs1bWzFNczFfCwBfSJdUTFwCYnIqOkYWADN6QjYAWgBHAFc4JRUuACEAQQBhAGlxKSQQOQUS1UaNBH8DZl9jX10Q7SNbHV8XdwRzTWZQ0Ls562MxXWjY9Gw8IlJKGgYmZkzssnyitmUSXkFRSTVmxXO1DttjcxnjXVNfTVMgvXNjbTjDy+ZgA-yBYKhcLmVbgdYJLbJXZpVhgagqCAnYrncrVOo3Rp3VqPRC6F7MSzmezmPwRMQ+IGTbSGKZ9Wx2UxiPS6AywuIbRLbFJ7FiwNEYrFnUr8YT1W7ye4qEmTMLMMTmMKaQLGAxU3xjNxaCLMPSadWjek-XTaaIxEAkWQQOBqfkIpI7VJMeUtB7tRA5AyMgN8+Gbd3ClFsDhkb2KtqgJ4GiaaD4Ul5U61iPrmTkwu2usNC5H7Q65QrYpWEhXEv0IbTaYzMXS6cw88y9YxmRyMzQGLwNjXszzZ0yaEPxQtIz2i8UkTHlqXxprV30JxCckE6dlciIa0xDHsGEH-QeZkdj-OhwVTkUHRhoHCxmtrhBmqlq3xWAFcnQfQ-TGws07IITH6FZbSAA */
```
Comment (alanplatt) [inline src/config/machines/example-grant-machine.js]: Would be a good to load these from a file per grant eventually....

Comment (Andrew-Folga) [reply src/config/machines/example-grant-machine.js]: Definitely in a future story in irritation 2 most likely together with machine per user session caching config.

File: src/utils/get-info-from-url.js
Comment (DavidBarker-EE) [inline src/utils/get-info-from-url.js:1]: This has found its way back in. All this data is available directly from the routers request object.

Comment (Andrew-Folga) [reply src/utils/get-info-from-url.js:1]: Thanks, addressed now.

**Review by DavidBarker-EE — APPROVED** [review]

File: src/views/400.njk [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{% extends 'layout.njk' %}
+{% block pageTitle %}
+  {{siteTitle}} - Bad request 
+{% endblock %}
+{% block content %}
+  <div class="govuk-width-container">
+    <div class="govuk-main-wrapper--auto-spacing">
+      <div class="govuk-grid-row">
+        <div class="govuk-grid-column-two-thirds">
+          <h1 class="govuk-heading-xl">Bad request.</h1>
```
Comment (swdpcomputing) [inline src/views/400.njk:10]: Remove period (isn't in the equivalent place on Page not Found).

Comment (Andrew-Folga) [reply src/views/400.njk:10]: I would leave it for now and fix it next time for this long standing PR if you agree.

**Pull request merged**